Add migration notes for subtitles as tracks

Arnei opened the pull request:
*This PR aims at contributing to the implementation of #4407, but touches on just one of the aspects to realize #4407's vision. It should therefore likely not be merged alone, but alongside other PRs.*

Adds some release notes about subtitles the changes to subtitles and how to migrate exisiting subtitles to conform to the changes. Put it in release notes since the migration is not just a script everyone can run and be done with, but will vary by institution.

Also adds a new WOH for migration. It allows to change the type of MediaPackageElements, e.g. from attachment to track.

Arnei committed:
Add migration notes for subtitles as tracks

Adds some release notes about subtitles as tracks and how to migrate.

Also adds a new WOH for migration. It allows to change the type of
mediapackages, e.g. from attachment to track.

Arnei committed:
Addendum to subtitle migration release notes - republish

Indicate that subtitles should be republished after converting them
to tracks from attachments.

miesgre commented on docs/guides/admin/docs/releasenotes/subtitles-as-first-class-citizens.txt:
```diff
@@ -0,0 +1,87 @@
+With Opencast 14 we want to put more emphasis on subtitles. You can find more details on how subtitles should be
+handled going forward in [Subtitles](../modules/subtitles.md).
+
+This comes with a bit of migration. Namely, subtitles should not be stored as "attachments" anymore, but as "media"
+(as they is called in the Admin UI) or "tracks" (as the are called internally). Therefore, all subtitle files currently
+stored as attachments in your events should be moved to tracks. This can easily be accomplished with the "changetype"
```
```suggestion
This comes with a bit of migration. Namely, subtitles should not be stored as "attachments" or "catalogs" anymore, but as "media"
(as they is called in the Admin UI) or "tracks" (as the are called internally). Therefore, all subtitle files currently
stored as attachments or catalogs in your events should be moved to tracks. This can easily be accomplished with the "changetype"
```

Arnei committed:
Update docs/guides/admin/docs/releasenotes/subtitles-as-first-class-citizens.txt

Co-authored-by: Miguel Escriva <miesgre@gmail.com>

mliradelc commented on docs/guides/admin/docs/workflowoperationhandlers/changetype-woh.md:
```diff
@@ -0,0 +1,45 @@
+Clone Workflow Operation
+========================
```
Should say `Change` instead of `Clone`?

Arnei replied:
Should always say change, thanks for the review.

mliradelc commented on docs/guides/admin/docs/workflowoperationhandlers/changetype-woh.md:
```diff
@@ -0,0 +1,45 @@
+
+Notes:
+
+- *source-flavor* and *source-tags* may be used both together to select media package elements based on both flavors and
+  tags If *source-flavor* is not specified, all media package elements matching *source-tags* will be selected
+- In case that neither *source-flavor* nor *source-tags* are specified, the operation will be skipped
+- In case no media package elements match *source-flavor* and *source-tags*, the operation will be skipped
+
```
Needs to show the valid track types. What types you can use?

mliradelc replied:
Here should be described that the types are `catalog` , `attachment` and `track`?

mliradelc commented on docs/guides/admin/docs/workflowoperationhandlers/changetype-woh.md:
```diff
@@ -0,0 +1,45 @@
+
+|Configuration Key         |Description                                       |Example           |
+|--------------------------|--------------------------------------------------|------------------|
+|source-flavors            |The source flavor(s) to clone                     |presenter/source  |
```
To change or to clone?

mliradelc reviewed (changes requested):
Seems very good, I have only to try the WoH.

Only I have some questions over the documentation.

Arnei committed:
Fix wording in changetype-woh.md

Wrongly said something about cloning, which is not what is happening.

Arnei committed:
Note valid types in changetype-woh.md

In theory there are more valid types, but these are the ones we
want to support.

Arnei committed:
Fix checkstyle in Apereo header

mliradelc commented:
@karendolan Are you using subtitles as catalogue? How do you see this migration tools?

mliradelc commented on docs/guides/admin/docs/releasenotes/subtitles-as-first-class-citizens.txt:
```diff
@@ -0,0 +1,87 @@
+This comes with a bit of migration. Namely, subtitles should not be stored as "attachments" or "catalogs" anymore, but as "media"
+(as they is called in the Admin UI) or "tracks" (as the are called internally). Therefore, all subtitle files currently
+stored as attachments or catalogs in your events should be moved to tracks. This can easily be accomplished with the "changetype"
+workflow operation handler new to Opencast 14. See example below. (Subtitles should then be republished)
```
`Opencast 15.x` instead.

mliradelc replied:
```suggestion
workflow operation handler new to Opencast 15. See example below. (Subtitles should then be republished)
```

mliradelc commented on docs/guides/admin/docs/releasenotes/subtitles-as-first-class-citizens.txt:
```diff
@@ -0,0 +1,87 @@
+- With OC14 we change a lot how subtitles work in Opencast, you may need to migrate!
```
```suggestion
- With OC 15 we change a lot how subtitles work in Opencast, you may need to migrate!
```

mliradelc commented on docs/guides/admin/docs/releasenotes/subtitles-as-first-class-citizens.txt:
```diff
@@ -0,0 +1,87 @@
+
+Subtitles as first class citizens
+---------------------------------------
+With Opencast 14 we want to put more emphasis on subtitles. You can find more details on how subtitles should be
```
```suggestion
With Opencast 15 we want to put more emphasis on subtitles. You can find more details on how subtitles should be
```

mliradelc reviewed (changes requested):
Few typos. The rest is ok.

Arnei committed:
Replace 14 with 15

Changes didn't make it into 14, so now 15 is the new goal

mliradelc reviewed (approved):
Approved!
